src/jit/codegenxarch.cpp

```diff
@@ -7424,20 +7427,42 @@ unsigned CodeGen::getBaseVarForPutArgStk(GenTreePtr treeNode)
 //
 // Returns: true if the stack pointer was adjusted; false otherwise.
 //
-bool CodeGen::genAdjustStackForPutArgStk(GenTreePutArgStk* putArgStk, bool isSrcInMemory)
+bool CodeGen::genAdjustStackForPutArgStk(GenTreePutArgStk* putArgStk)
 {
-    assert(isSrcInMemory || (putArgStk->gtOp1->OperGet() == GT_FIELD_LIST));
-
-    // If this argument contains any GC pointers or is less than 16 bytes in size and is either in memory or composed
-    // entirely of slot-like fields (i.e. integral-types, 4-byte-aligned fields that take up 4 bytes including any
-    // padding), we will use a sequence of `push` instructions to store the argument to the stack.
     const unsigned argSize = putArgStk->getArgSize();
-    if ((putArgStk->gtNumberReferenceSlots != 0) ||
-        ((argSize < 16) && (isSrcInMemory || (putArgStk->gtPutArgStkKind == GenTreePutArgStk::Kind::AllSlots))))
+
+// If the gtPutArgStkKind is one of the push types, we do not pre-adjust the stack.
+// This is set in Lowering, and is true if and only if:
+// - This argument contains any GC pointers OR
+// - It is a GT_FIELD_LIST OR
+// - It is less than 16 bytes in size.
+#ifdef DEBUG
```

Comment:
You could avoid clang-format messing with the indenting here by using CLANG_FORMAT_COMMENT_ANCHOR;